source/common/ssl/connection_impl.cc

```diff
@@ -139,16 +135,25 @@ Network::ConnectionImpl::PostIoAction ConnectionImpl::doHandshake() {
 
 void ConnectionImpl::drainErrorQueue() {
   bool saw_error = false;
+  bool saw_counted_error = false;
```

Comment:
All errors from the `ContextImpl::verifyCallback()` will result in  `SSL_R_CERTIFICATE_VERIFY_FAILED`, so the current approach should be robust enough.

I've tried implementing your suggestion, but (a) we still need to test for `SSL_R_PEER_DID_NOT_RETURN_A_CERTIFICATE` here, so we don't gain much by removing the other check, (b) the flag would need to be accessed from both `ConnectionImpl` and `ContextImpl`, so it requires extra boilerplate to expose.

I'm going to leave it as-is, but thanks for the suggestion.